### PR TITLE
feat(modeler): new image-selection modal

### PR DESCRIPTION
**Pull request by xanpj**
Relevant for https://github.com/pinussilvestrus/postit-js/issues/17

**New Image-Selection Modal.** 
URL sets the image source from a url. Local upload is converting to base64 for demonstration purposes. Obviously, base64 images are slow for usage on the canvas.

The upload itself is multi-upload ready but currently it cannot draw multiple images on upload to the board because of the placeholder logic.

I also moved some of the css code from the examples/css into the postit css code because the example buttons and modals are now part of the default design.

I also added a few helpers for file upload, mouse position retrieval etc.

Screen Shot 2020-06-06 at 6 54 37 PM

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
Why it is necessary, that the keybard helper modals and stuff are included in the core library? I think it should be up to the library user to use these additional components, to not blow the library up.

**Comment by pinussilvestrus** [comment]
I added you as a contributor of this project. Are you able to create branches instead of forking the project?

**Comment by xanpj** [comment]
1) Great!
2) I might have ported too much of the example css code. What I have in mind is porting the buttons and modal classes into the postit css because the new modal and its ui make use of them. The Modal and Button css would then become the default design.
3) Absolutely. Base64 is no real solution. But if someone starts to use postit with a backend he will already have the proper ui for the local file upload. I assume the best solution is to give a warning sign next to the local file upload.

**Comment by pinussilvestrus** [comment]
2. Yeah that makes sense! Let's only move buttons and modals css to the core styles and keep the other stuff in the Demo
3. I think adding a warning and/or a beta indicator next to the file upload would be good enough as a first iteration :+1:

**Comment by pinussilvestrus** [comment]
Can we prefix all css classes we use in the core with `pjs-`? Just to make it consistent.

File: assets/postit-js.css [hunk 1]
```diff
@@ -90,4 +167,306 @@
   background-color: black !important;
 }
 
+/* screen interaction helpers & modal */
+
+.io-editing-tools {
```
Comment (pinussilvestrus) [inline assets/postit-js.css:172]: Can we prefix all css classes we use in the core with `pjs-`? Just to make it consistent. Also with the buttons and so on above

**Comment by xanpj** [comment]
1) Css is organized so that only framework css is in the postit-js.css
2) All css is prfixed with pjs- (or djs- if it overwrites diagram.js style)
3) Added a warning icon with info tooltip next for the local file upload
4) Put all copy text in translate functions (except plural s in "files", but this Im leavning for a more general solution)

Screen Shot 2020-06-08 at 3 18 06 PM

**Comment by xanpj** [comment]
As a final addition: I fixed the plural text issue, and removed unnecessary error behaviour.

**Comment by pinussilvestrus** [comment]
Thanks for the update! I'll have a look at the code soon 👍

**Comment by pinussilvestrus** [comment]
I updated some tweaks in the code and merged the changes with https://github.com/pinussilvestrus/postit-js/commit/f57b1e84e3c37795265048d3afe13f2848d216c5. Thanks! I think a good improvement 👏

**Pull request closed without merging**